mhkit/wave/graphics.py

```diff
@@ -294,3 +294,49 @@ def plot_chakrabarti(H, lambda_w, D, ax=None):
     ax.set_ylabel('KC parameter, $\\frac{H}{D}$')
 
     plt.tight_layout()
+
+def plot_environmental_contour(x1, x2, x1_contour, x2_contour, 
+                               data_label=None, 
+                               contour_label=None,
+                               ax=None):
+    """
+    Plots an overlay of the x1 and x2 variables to the calculated
+    contours.
+
+    Parameters
+    ------------
+    x1: array like
+        Component 1 data
+    x2: array like
+        Component 2 data     
+    x1_Return : np.array
+        Calculated x1 values along the contour boundary following
+        return to original input orientation.
+    x2_Return : np.array
+       Calculated x1 values along the contour boundary following
+        return to original input orientation.
+    data_label: string
+        Label for x1, x2 data
+    contour_label: string
+        Label for countor data (x1_contour, x2_contour)    
+    ax : matplotlib axes object
+        Axes for plotting.  If None, then a new figure is created.
+
+    Returns
+    ---------
+    ax : matplotlib pyplot axes
+
+    """
+
+    #assert isinstance(eta, pd.DataFrame), 'eta must be of type pd.DataFrame'
+    
+
+    ax = _xy_plot(x2_contour, x1_contour, 'k-', 
+                  label=contour_label, ax=ax)
+    ax = plt.plot(x2, x1, 'bo', alpha=0.1, 
+                  label=data_label) 
+    plt.legend(loc='lower right')
+    plt.xlabel('Energy period, $T_e$ [s]')
```

Comment:
The input variables are generic for this function, yet the x and y labels are explicit. Will users use this function for parameters other than Hm0 and Te? If so, we should make the x and y labels optional input. If not, we should have more explicit input parameter names.

Reply:
Yeah I'm pretty torn on this. One of the major questions on WDRT is "can I use variable "x" to ..." and the answer is always "It is a statistical method, you can pass any variables you want".  I guess we can use a nondescipt "T" for the x-axis and Hs for the y-axis. lmk if you have thoughts on how to better generalize this

Reply:
@rpauly18 @ssolson I personally think it should be generalized to develop a contour for any two provided variables. For WECs the obvious use is T vs Hs, but what if you wanted to know Tp vs wind_speed, or Hs vs current, these could be applicable to other MRE applications and offshore wind.